pkg/rancher-desktop/config/settings.ts

```diff
@@ -474,114 +473,142 @@ function updateSettings(settings: Settings) {
+ * defaults or locked profile, the user directory will not be read.
+ * @returns undefined, or type validated defaults and locked deployment profiles.
  */
-export function readDeploymentProfile() {
-  let profileData;
-  let profile;
+export function readDeploymentProfiles() {
+  let defaults;
+  let locked;
 
   switch (os.platform()) {
   case 'win32':
-    try {
-      profile = readRegistryDeploymentProfile(); // TODO: call with path to system profile
-    } catch {
-      try {
-        profile = readRegistryDeploymentProfile(); // TODO: call with path to user profile
-      } catch {}
+    for (const key of [nativeReg.HKLM, nativeReg.HKCU]) {
+      const registryKey = nativeReg.openKey(key, REGISTRY_PATH_PROFILE.join('\\'), nativeReg.Access.READ);
+
+      if (registryKey !== null) {
+        defaults = readRegistryUsingSchema(null, defaultSettings, registryKey, ['Defaults']);
+        locked = readRegistryUsingSchema(null, defaultSettings, registryKey, ['Locked']);
+        nativeReg.closeKey(registryKey);
+        if (typeof defaults !== 'undefined' || typeof locked !== 'undefined') {
+          break;
+        }
+      }
     }
     break;
   case 'linux':
-    try {
-      profileData = fs.readFileSync(join(paths.deploymentProfileSystem, PROFILE_JSON), 'utf8');
-    } catch {
+    for (const rootPath of [paths.deploymentProfileSystem, paths.deploymentProfileUser]) {
       try {
-        profileData = fs.readFileSync(join(paths.deploymentProfileUser, PROFILE_JSON), 'utf8');
+        const defaultsData = fs.readFileSync(join(rootPath, 'defaults.json'), 'utf8');
+
+        defaults = JSON.parse(defaultsData);
       } catch {}
-    }
-    if (typeof profileData !== 'undefined') {
-      profile = JSON.parse(profileData);
+      try {
+        const lockedData = fs.readFileSync(join(rootPath, 'locked.json'), 'utf8');
+
+        locked = JSON.parse(lockedData);
+      } catch {}
+
+      if (typeof defaults !== 'undefined' || typeof locked !== 'undefined') {
+        break;
+      }
     }
     break;
   case 'darwin':
-    try {
-      profileData = fs.readFileSync(join(paths.deploymentProfileSystem, PROFILE_PLIST), 'utf8');
-    } catch {
+    for (const rootPath of [paths.deploymentProfileSystem, paths.deploymentProfileUser]) {
       try {
-        profileData = fs.readFileSync(join(paths.deploymentProfileUser, PROFILE_PLIST), 'utf8');
+        const defaultsData = fs.readFileSync(join(rootPath, 'io.rancherdesktop.profile.defaults.plist'), 'utf8');
+
+        defaults = plist.parse(defaultsData);
       } catch {}
-    }
-    if (typeof profileData !== 'undefined') {
-      profile = plist.parse(profileData);
+      try {
+        const lockedData = fs.readFileSync(join(rootPath, 'io.rancherdesktop.profile.locked.plist'), 'utf8');
+
+        locked = plist.parse(lockedData);
+      } catch {}
+
+      if (typeof defaults !== 'undefined' || typeof locked !== 'undefined') {
+        break;
+      }
```

Comment:
There's so much repetition here, I think you can write a function for the `darwin` and `linux` cases and pass in `['defaults.json', 'locked.json', JSON]` or `['io.rancherdesktop.profile.defaults.plist', 'io.rancherdesktop.profile.locked.plist', plist]`.  The function can be something like `readProfile(defaultsPath: string, lockedFieldsPath: string, parser: any)`

Reply:
ok, done.